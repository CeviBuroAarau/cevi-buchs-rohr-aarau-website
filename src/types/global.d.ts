declare let global: any; // eslint-disable-line @typescript-eslint/no-explicit-any
