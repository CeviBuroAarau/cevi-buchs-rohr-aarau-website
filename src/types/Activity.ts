export interface Activity {
  title: string;
  url: string;
  thumb: string;
}

export interface CockpitActivities {
  entries: ActivityRaw[];
}

export interface ActivityRaw {
  title: string;
  image: {
    path: string;
  };
  thumb: {
    path: string;
  };
}
