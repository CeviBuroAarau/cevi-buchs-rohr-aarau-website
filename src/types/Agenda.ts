export interface Agenda {
  title: string;
  text: string;
  date: Date;
}

export interface CockpitAgenda {
  entries: Agenda[];
}

export interface CockpitScope {
  display: string;
}

export interface CockpitEventInfo {
  scope: CockpitScope;
  text: string;
  date: Date;
}

export interface EventInfo {
  scope: string;
  text: string;
  date: Date;
}

export interface CockpitEventInfos {
  entries: CockpitEventInfo[];
}
