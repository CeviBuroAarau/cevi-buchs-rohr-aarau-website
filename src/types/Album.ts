export interface Album {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: AlbumImage[];
  previewImage: AlbumPreviewImage;
}

export interface AlbumImage {
  title: string;
  url: string;
}

export interface AlbumPreviewImage {
  url: string;
}

export interface CockpitAlbum {
  entries: CockpitAlbumEntry[];
}

export interface CockpitAlbumEntry {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: CockpitAlbumImage[];
  previewImage: CockpitAlbumPreviewImage;
}

export interface CockpitAlbumImage {
  meta: {
    title: string;
    asset: string;
  };
  path: string;
}

export interface CockpitAlbumPreviewImage {
  path: string;
}
