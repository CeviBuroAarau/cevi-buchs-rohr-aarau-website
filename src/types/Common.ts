export enum FormState {
  NotDisplayed,
  Displayed,
}
