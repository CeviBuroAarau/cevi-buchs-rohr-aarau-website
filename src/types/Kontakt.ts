export interface KontaktFormRequest {
  form: {
    name: string;
    email: string;
    message: string;
  };
}

export interface KontaktFormResponse {
  error: string;
  data: {
    name: string;
    email: string;
    message: string;
  };
}
