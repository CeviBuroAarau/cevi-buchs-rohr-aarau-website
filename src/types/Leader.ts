export interface Leader {
  name: string;
  isactive: boolean;
  scoutname: string;
  function: string[];
  group: string;
  birthyear: string;
  place: string;
  profession: string;
  recreation: string;
  inScoutsSince: string;
  inScoutsBecause: string;
  bestExperiences: string;
  file: string;
}

export interface LeaderRaw {
  name: string;
  isactive: boolean;
  scoutname: string;
  function: string[];
  group: {
    display: string;
  };
  birthyear: string;
  place: string;
  profession: string;
  recreation: string;
  inScoutsSince: string;
  inScoutsBecause: string;
  bestExperiences: string;
  image: {
    path: string;
  };
}

export interface CockpitLeaders {
  entries: LeaderRaw[];
}
