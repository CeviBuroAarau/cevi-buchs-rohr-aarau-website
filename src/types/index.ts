export * from "./Activity";
export * from "./Agenda";
export * from "./Album";
export * from "./Media";
