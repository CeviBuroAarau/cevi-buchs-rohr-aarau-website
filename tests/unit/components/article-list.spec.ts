import { shallowMount } from "@vue/test-utils";
import ArticleList from "@/components/article-list.vue";
import { Article } from "@/types";

describe("ArticleList Component", () => {
  const articles: Article[] = [
    {
      name: "Jungscharhemd Kinder",
      file:
        process.env.VUE_APP_COCKPIT_FILES +
        "\\/storage\\/uploads\\/2021\\/01\\/28\\/haupt_01_cevi_hemd.jpg_uid_6013250287435.webp",
      price: "54.9",
      categories: ["Regul\\u00e4re Artikel"],
      description: "Verf\\u00fcgbare Gr\\u00f6ssen 140, 152, 164",
    },
    {
      name: "Jungscharhemd Erwachsene",
      file:
        process.env.VUE_APP_COCKPIT_FILES +
        "\\/storage\\/uploads\\/2021\\/01\\/28\\/haupt_02_cevi_hemd_erwachsene.jpg_uid_60132502b1bcf.webp",
      price: "62.9",
      categories: ["Regul\\u00e4re Artikel"],
      description: "Verf\\u00fcgbare Gr\\u00f6ssen S, M, L, XL, 2XL",
    },
  ];

  test("render article-list", () => {
    const wrapper = shallowMount(ArticleList, {
      stubs: [],
      propsData: {
        articles: articles,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
