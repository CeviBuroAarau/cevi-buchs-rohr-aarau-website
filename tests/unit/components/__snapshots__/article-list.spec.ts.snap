// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render article-list 1`] = `
<ul>
  <li
    class="vue-light-gallery-thumb"
  >
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        <p
          class="card-header-title is-centered"
        >
          
          Jungscharhemd Kinder
        
        </p>
      </header>
       
      <div
        class="card-image"
      >
        <img
          alt=""
          src="https://backend.cevi-buro-aarau.ch/\\\\/storage\\\\/uploads\\\\/2021\\\\/01\\\\/28\\\\/haupt_01_cevi_hemd.jpg_uid_6013250287435.webp"
        />
      </div>
       
      <div
        class="card-content"
      >
        <div
          class="content"
        >
          
          54.9 CHF
          <br />
          
          Verf\\u00fcgbare Gr\\u00f6ssen 140, 152, 164
        
        </div>
      </div>
    </div>
  </li>
  <li
    class="vue-light-gallery-thumb"
  >
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        <p
          class="card-header-title is-centered"
        >
          
          Jungscharhemd Erwachsene
        
        </p>
      </header>
       
      <div
        class="card-image"
      >
        <img
          alt=""
          src="https://backend.cevi-buro-aarau.ch/\\\\/storage\\\\/uploads\\\\/2021\\\\/01\\\\/28\\\\/haupt_02_cevi_hemd_erwachsene.jpg_uid_60132502b1bcf.webp"
        />
      </div>
       
      <div
        class="card-content"
      >
        <div
          class="content"
        >
          
          62.9 CHF
          <br />
          
          Verf\\u00fcgbare Gr\\u00f6ssen S, M, L, XL, 2XL
        
        </div>
      </div>
    </div>
  </li>
</ul>
`;
